R/project-rel.R

```diff
@@ -3,98 +3,121 @@
 #'
 #' @param .compiler A [substrait_compiler()] or object that can be coerced to one
 #' @param ... Expressions
+#' @param .drop_columns A character vector of columns to explicitly drop before
+#'   adding expressions specified in `...`.
 #'
 #' @return A modified `.compiler`
 #' @export
 #'
 #' @examples
-#' substrait_project(
-#'   data.frame(a = 1, b = "one"),
-#'   c = a + 1
-#' )
+#' substrait_select(data.frame(a = 1, b = "one"), c = a + 1)
 #'
-substrait_project <- function(.compiler, ...) {
+substrait_project <- function(.compiler, ..., .drop_columns = character()) {
   .compiler <- substrait_compiler(.compiler)$clone()
 
-  # evaluate expressions sequentially, updating the compiler as we go so that
+  # Evaluate expressions sequentially, updating the compiler as we go so that
   # fields created by earlier arguments are accessible from later arguments
   quos <- rlang::enquos(..., .named = TRUE)
   expressions <- list()
   types <- list()
 
+  # Keep a list of columns that we need to drop as part of this relation
+  drop_columns <- character()
```

Comment:
I think having an argument named `.drop_columns` and then this variable names `drop_columns` is a little confusing - mind renaming this?